proto/cacheclient.proto

```diff
@@ -402,16 +402,89 @@ message _ListRemoveRequest {
 
 message _ListRemoveResponse {}
 
+// List fetch + retain index behavior
+// This follows redis behavior for the most part, differences
+// 1. end_index is exclusive, unless None
+// 2. `start_index` == 0, then we start at the front of list
+// 2. `end_index` == None, start at the end of the index, inclusive
+/*
+  list_ = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
+
+  # Using the slice notation:
+  # start:stop(:step)?
+
+
+  # start=None, end=None
+  list_[:] == [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
+
+
+  # start=5, end=None
+  list_[5:] == [5, 6, 7, 8, 9]
+
+  # start=5, end=0
+  list_[5:0] == [5, 6, 7, 8]
+
+  # start=-5, end=None
+  list_[-5:] == [5, 6, 7, 8, 9]
+
+  # start=-5, end=0
+  list_[-5:0] == [5, 6, 7, 8]
```

Comment:
Same thing here: start=-5, end=-1 (up to but not including the final list element)